stix_shifter_modules/nozomi/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,549 @@
+                        query_list = self._check_max_query_length(row_01, row_02, combined_query, query_list)
+                    else:
+                        combined_query = f'{row_01} {operator} {row_02.replace("where ", "")}'
+                        query_list = self._check_max_query_length(row_01, row_02, combined_query, query_list)
+                else:
+                    if row_02 not in query_list:
+                        query_list.append(row_02)
+            if not combined_flag:
+                if row_01 not in query_list:
+                    query_list.append(row_01)
+        return query_list
+
+    def combine_and_queries(self, expression_01, expression_02, operator) -> list:
+        """
+        combine the queries using and operator
+        :param expression_01: expression object
+        :param expression_02: expression object
+        :param operator: string
+        :return query list
+        """
+        query_list = []
+
+        for row_01 in expression_01:
+            for row_02 in expression_02:
+                combined_query = f'{row_01} {operator} {row_02}'
+                if len(combined_query) > MAX_QUERY_LENGTH and not self.logged:
+                    self.logged = True
+                    logger.info('Unable to split the query. Query length is more than maximum length. '
+                                'Use OR operator to split the query.')
+                query_list.append(combined_query)
+        return query_list
+
+    def _eval_combined_comparison_exp(self, expression) -> str:
+        """
+        Function for parsing combined comparison expression
+        :param expression: expression object
+        """
+        query = []
+        operator = self._lookup_comparison_operator(expression.operator)
+        expression_01 = self._parse_expression(expression.expr1)
+        expression_02 = self._parse_expression(expression.expr2)
+
+        if not expression_01 or not expression_02:
+            return query
+
+        if expression.operator == ComparisonExpressionOperators.Or:
+            query = self.combine_or_queries(expression_01, expression_02, operator)
+        elif expression.operator == ComparisonExpressionOperators.And:
+            query = self.combine_and_queries(expression_01, expression_02, operator)
+        return query
+
+    def _eval_combined_observation_exp(self, expression, qualifier=None) -> str:
+        """
+        Function for parsing combined observation expression
+        :param expression: expression object
+        :param qualifier: qualifier
+        """
+        expression_01 = self._parse_expression(expression.expr1, qualifier)
+        expression_02 = self._parse_expression(expression.expr2, qualifier)
+        operator = self._lookup_comparison_operator(expression.operator)
+
+        query = []
+        if expression_01 and expression_02:
+            query = self.combine_or_queries(expression_01, expression_02, operator)
```

Comment:
This does not work for 

"[ipv4-addr:value = '1.1.1.1'] AND [ipv4-addr:value = '1.1.1.1']"
the result is
"query=alerts | where ip_dst==\"1.1.1.1\" OR ip_dst==\"1.1.1.1\" | where record_created_at>=1712258269039 | where record_created_at<=1712258569039"

which is an OR command.

Reply:
It is as designed. AND operator between two observation statements is converted to OR. All connectors are following the same.

Reply:
After speaking with @mdazam1942, my understanding is that the observations should be treated as separate queries. However, given that doing so effectively treats all observations as OR's, what you've done is probably the same thing. 

I'll let @mdazam1942 clarify as I'm not 100% here.

Reply:
Multiple Observations with OR/AND operator are translated to multiple queries if timestamp is different for each observation. However if timestamp is same then effectively a single data source query with OR operator is created.